dependency_cache.go

```diff
@@ -370,19 +381,27 @@ func (DependencyCache) verify(path string, expected string) error {
 	return nil
 }
 
-func (d DependencyCache) overrideDependencySource(urlD *url.URL) {
-	dependencySourceOverride := sherpa.GetEnvWithDefault("BP_DEPENDENCY_SOURCE_OVERRIDE", "")
-	if dependencySourceOverride != "" {
-		d.Logger.Infof("variable BP_DEPENDENCY_SOURCE_OVERRIDE found. overriding download uri.")
-		urlOverride, err := url.ParseRequestURI(dependencySourceOverride)
-		if err == nil {
+func (d DependencyCache) setDependencyMirror(urlD *url.URL) {
+	dependencyMirror := sherpa.GetEnvWithDefault("BP_DEPENDENCY_MIRROR", "")
+	dependencyMirrorPreserveHost := sherpa.GetEnvWithDefault("BP_DEPENDENCY_MIRROR_PRESERVE_HOST", "false")
```

Comment:
These settings should be moved to the DependencyCache struct.

I would suggest you then update `NewDependencyCache` such that it pulls from env variables (i.e. default behavior). That will mean it only pulls from the env variables once, instead of every download.

I think it will make testing easier too. You won't need to set the env variables, you can just make a new struct and set the config values.

Reply:
Also, I would suggest `sherpa.ResolveBool` or `sherpa.ResolveBoolErr` here instead, that way you have a bool to work with and you don't need string comparisons.

https://github.com/paketo-buildpacks/libpak/blob/main/sherpa/env_var.go#L57